tests/ui/test_pv_encryption_ui.py

```diff
@@ -0,0 +1,287 @@
+            vol_handle = pv_obj.get().get("spec").get("csi").get("volumeHandle")
+            logger.info(f"Volume handle {vol_handle}")
+            vol_handles.append(vol_handle)
+            logger.info(f"Volume handles {vol_handles}")
+            logger.info("key is created in Vault verified")
+
+            # Check if encryption key is created in Vault
+            logger.info("Checking if encryption key is created in Vault")
+            if kms.is_key_present_in_path(
+                key=vol_handle, path=self.vault.vault_backend_path
+            ):
+                logger.info(f"Vault: Found key for {pvc[0].name}")
+            else:
+                raise ResourceNotFoundError(f"Vault: Key not found for {pvc[0].name}")
+
+        ocp_version = get_ocp_version()
+        self.pvc_loc = locators[ocp_version]["pvc"]
+
+        # Verify whether encrypted device is present inside the pod and run IO
+        logger.info(
+            "Verify whether encrypted device is present inside the pod and run IO"
+        )
+        for vol_handle, pod_obj in zip(vol_handles, pod_objs):
+            if pod_obj.exec_sh_cmd_on_pod(
+                command=f"lsblk | grep {vol_handle} | grep crypt"
+            ):
+                logger.info(f"Encrypted device found in {pod_obj.name}")
+            else:
+                logger.error(f"Encrypted device not found in {pod_obj.name}")
+
+            logger.info(f"Running FIO on Pod '{pod_obj.name}'")
+            pod_obj.run_io(
+                storage_type="block",
+                size=(int(pvc_size) - 1),
+                io_direction="write",
+                runtime=60,
+                invalidate=0,
+            )
+            logger.info(f"IO started on the Pod {pod_obj.name}")
+
+        # Wait for IO completion
+        logger.info("Waiting for IO completion on the Pod")
+        for pod_obj in pod_objs:
+            pod_obj.get_fio_results()
+        logger.info("IO completed on all Pods")
+
+        # Delete the pod
+        for pod_obj in pod_objs:
+            logger.info(f"Deleting Pod '{pod_obj.name}'")
+            pod_obj.delete(wait=True)
+            pod_obj.ocp.wait_for_delete(resource_name=pod_obj.name)
+            logger.info(f"Pod '{pod_obj.name}' Deleted Successfully")
+
+        # Deleting the PVC via UI
+        logger.info(f"Delete pvc '{pvc_name}' via UI")
+        pvc_objs = get_all_pvc_objs(namespace=project_name)
+        pvcs = [pvc_obj for pvc_obj in pvc_objs if pvc_obj.name == pvc_name]
+
+        for pvc in pvc_objs:
+            pvc_ui_obj.delete_pvc_ui(pvc.name, project_name)
+            logger.info(f"Waiting for '{pvc.name}' pvc deletion")
+            pvcs[0].ocp.wait_for_delete(pvc.name, timeout=120)
+
+        logger.info("Checking if pvc's are deleted or not")
+        if len(pvcs) > 0:
+            assert f"PVC {pvcs[0].name} does not deleted"
+
+        # Verify whether the key is deleted in Vault. Skip check for kv-v2 due to BZ#1979244
```

Comment:
It would be good to add teardown/cleanup for the vault resources (backend path and namespace) that have been created as part of the test and also any changes that have been made to the csi-kms-connection-details configmap should be reverted.